### PR TITLE
Merging 'assets' with 'pelican-webassets'

### PR DESCRIPTION
**Pull request by bryanbrattlof**
Hi Everyone!

There is a lot in here so here's the breakdown. I was having too much trouble with merge conflicts while trying to cherry pick changes from `pelican-webassets` so I started from scratch. There are three main sections to this PR:

- I pulled the existing tests out of the package and revived them back into working order. I also added a `poetry run pip install --upgrade libsass` line into the `tests` workflow so all of the (now 17) tests will pass.

- deprecated the `ASSET_*` options and combined them with the `WEBASSETS_*` options so they both will continue to work while people move over to using the new `WEBASSETS_*` option naming. If people are using the `ASSET_*` options a `WARNING` will be emitted. This is where the bulk of the new tests come from.

- I refactored the README to replace the `ASSET_*` naming along with breaking up the README into the customary `Installation` `Usage` and `Configuration` sections, adding in (hopefully useful) example setups and links to relevant documentation.

I hope everyone is having a great holiday season!

### CODE REVIEW

File: pelican/plugins/webassets/webassets.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 .. _webassets: https://webassets.readthedocs.org/
 
 """
-from __future__ import unicode_literals
+# from __future__ import unicode_literals
```
Comment (kdeldycke) [inline pelican/plugins/webassets/webassets.py]: Feel free to remove that `import` entirely: this plugin is [no longer Python 2.x compatible](https://github.com/pelican-plugins/webassets/blob/856cfa4b5d8cfeee5916909dc480ccbff737d204/pyproject.toml#L37).

File: pelican/plugins/webassets/webassets.py [hunk 2]
```diff
@@ -47,33 +47,67 @@ def create_assets_env(generator):
     assets_destination = os.path.join(generator.output_path, theme_static_dir)
     generator.env.assets_environment = Environment(assets_destination, theme_static_dir)
 
-    if "ASSET_CONFIG" in generator.settings:
-        for item in generator.settings["ASSET_CONFIG"]:
-            generator.env.assets_environment.config[item[0]] = item[1]
+    # TODO: remove deprcated variables in 2022
```
Comment (kdeldycke) [inline pelican/plugins/webassets/webassets.py]: Typo: `deprcated` -> `deprecated`.

File: tests/test_webassets.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        generator = self.get_generators({
+            'ASSET_SOURCE_PATHS': source_paths})
+
+        paths = generator.env.assets_environment.load_path
+
+        for path in self.settings['THEME_STATIC_PATHS'] + source_paths:
+            self.assertIn(str(Path(generator.theme, path)), paths)
+
+
+class TestDepricationDate(unittest.TestCase):
```
Comment (kdeldycke) [inline tests/test_webassets.py:311]: Ahaha, that's a fun one! 😄 Great way to keep a reasonable time-frame for the migration window! 😎

**Review by kdeldycke — COMMENTED** [review]
Thanks a lot @bryanbrattlof for that PR. I'm impressed by the level of details and care about that little plugin.

Appart for inconsequential remarks in my review, this PR is good to go. Feel free to merge as soon as you want! 👍

**Review by kdeldycke — APPROVED** [review]

File: pelican/plugins/webassets/webassets.py (hunk 1 above)
Comment (bryanbrattlof) [reply pelican/plugins/webassets/webassets.py]: Thanks, I totally didn't notice that! I've amended this to the c51b978 **- make poetry's linters happy** commit

File: pelican/plugins/webassets/webassets.py (hunk 2 above)
Comment (bryanbrattlof) [reply pelican/plugins/webassets/webassets.py]: Haha one day I'll learn how to spell that word or get better at [flyspell](https://www.emacswiki.org/emacs/FlySpell)

I've amended this to the 424c27c **- deprecate ASSET_CONFIG in favor of WEBASSETS_CONFIG** commit

**Comment by bryanbrattlof** [comment]
Thanks for the review @kdeldycke 

I don't seem to have merge permissions ¯\\_(ツ)_/¯ So I'll leave the merging up to you

Thanks again for giving the time to review the PR

**Comment by kdeldycke** [comment]
Move fast and merge things! 😁

Thanks again @bryanbrattlof for your outstanding contribution! 🏅

**Pull request merged**